src/bluesky/tests/test_flyer.py

```diff
@@ -623,3 +625,109 @@ def fly_plan():
     assert flyer2.call_counts["kickoff"] == 1
     assert flyer1.call_counts["complete"] == 1
     assert flyer2.call_counts["complete"] == 1
+
+
+def collect_data_during_multiple_scan(dets, motors, stream_name: str = "test_stream"):
+    yield from open_run()
+    yield from declare_stream(*dets, name=stream_name, collect=True)
+
+    grid_scan: list = [10, 10, 10]
+
+    yield from prepare(*dets, sum(grid_scan))
```

Comment:
I was expecting to see this prepare the dets for `[10, 10, 10]`, then call kickoff & complete 3 times, otherwise we don't know whether the detectors have produced all the frames needed for a particular inner scan.

Thinking about it, we I don't think my use case 1 in https://github.com/bluesky/bluesky/issues/1849 is quite correct, let's discuss in person...